terraform/modules/hub/s3.tf

```diff
@@ -22,3 +29,17 @@ resource "aws_s3_bucket" "deployment_config" {
     }
   }
 }
+
+resource "aws_s3_bucket" "verify_frontend_assets" {
+  bucket = "gds-${var.deployment}-verify-frontend-app-assets-${random_string.verify_frontend_assets_bucket_name_suffix.result}"
+  acl    = "private"
```

Comment:
If we want to restrict people going directly to the bucket we would probably need to follow something like this: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/PrivateContent.html

However, since we're talking about CSS/js/etc we could probably go simpler with a `public-read` policy and also give that policy on the objects: 
https://docs.aws.amazon.com/AmazonS3/latest/dev/example-bucket-policies.html#example-bucket-policies-use-case-2

We should also configure CORS headers on the bucket: https://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html

Reply:
If we set the _whole bucket_ to `public-read`, do we need a `s3:GetObject` policy as well? That seems counter-intuitive to me?

Reply:
Done the public read policy on the objects in  e0f1b46.

Do we have prior art for CORS anywhere?

Reply:
I'm just thinking right now that this might be better done in `verify-terraform` in `environments/tools`; we aren't going to build and ship the assets into multiple environments. They can all reference the same bucket...

Reply:
Aren't we? What if we want to test asset changes in eg joint but run a different version in production?

Reply:
yeah but I was thinking our assets would be accessible on the path:
 `s3://name-of-bucket/${verify_frontend_commit_hash}-${verify_frontend_federation_config}/`

then we don't have to do promotion but it doesn't go to all environments at once, then the version number is passed in

Reply:
i.e. we have a `latest` and a `$commit_hash` bucket

Reply:
my thinking was that we'd do some asset bundling stuff in the Concourse job and push the objects to S3 on deploy. So they'd be `s3://name-of-bucket/asset1-randomrailshash.png" or somesuch (I've not yet thought about this extensively), so when we deploy joint we'd have those assets, then the same process would happen again when we deploy staging, etc. etc. So every environment would have a separate bucket (like we do with Prometheus configs, for instance).

Reply:
@tlwr I think it would be good to pair on this. I've a) never touched `verify-terraform`, b) it's a Verify repo and c) I don't want to get wires crossed, so it would be good to get this out together in the intended way.